src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -433,39 +443,40 @@ public synchronized void unsubscribe(UnsubscribeRequest request)
         return CompletableFuture.completedFuture(0);
     }
 
-    private synchronized void spoolMessage()  {
-        if (spoolingFuture.get() == null || spoolingFuture.get().isCancelled()) {
-            spoolingFuture.set(ses.scheduleWithFixedDelay(this::spoolTask, 0, 5, TimeUnit.SECONDS));
+    protected CompletableFuture<Integer> spoolSingleMessage() throws InterruptedException {
+        final long id = spool.popId();
+        PublishRequest request = spool.getMessageById(id);
+        if (request == null) {
+            return null;
         }
+        MqttMessage m = new MqttMessage(request.getTopic(), request.getPayload());
+        return getConnection(false).publish(m, request.getQos(), request.isRetain())
+                .whenComplete((packetId, throwable) -> {
+                    // packetId is the SDK assigned ID. Ignore this and instead use the spooler ID
+                    if (throwable == null) {
+                        spool.removeMessageById(id);
+                        logger.atDebug().kv("id", id).kv("topic", request.getTopic())
+                                .log("Successfully published message");
+                    } else {
+                        spool.addId(id);
+                        logger.atError().log("Failed to publish the message via Spooler", throwable);
+                    }
+                });
     }
 
     /**
      * Iterate the spooler queue to publish all the spooled message.
      */
-    protected void spoolTask() {
+    protected void spoolMessages() {
```

Comment:
Like we talked about, rename this since it isn't spooling messages, this is taking messages off of the spool. Maybe something like `publishTask`